operator/main.go

```diff
@@ -398,42 +399,15 @@ func onOperatorStartLeading(ctx context.Context) {
 			log.WithError(err).Fatalf("Unable to start %s allocator", ipamMode)
 		}
 
-		startSynchronizingCiliumNodes(nm)
-		nodeManager = &nm
-
-		switch ipamMode {
-		case ipamOption.IPAMClusterPool:
-			// We will use CiliumNodes as the source of truth for the podCIDRs.
-			// Once the CiliumNodes are synchronized with the operator we will
-			// be able to watch for K8s Node events which they will be used
-			// to create the remaining CiliumNodes.
-			<-k8sCiliumNodesCacheSynced
-
-			// We don't want CiliumNodes that don't have podCIDRs to be
-			// allocated with a podCIDR already being used by another node.
-			// For this reason we will call Resync after all CiliumNodes are
-			// synced with the operator to signal the node manager, since it
-			// knows all podCIDRs that are currently set in the cluster, that
-			// it can allocate podCIDRs for the nodes that don't have a podCIDR
-			// set.
-			nm.Resync(context.Background(), time.Time{})
-		}
-	default:
-		startSynchronizingCiliumNodes(NOPNodeManager)
-		nodeManager = &NOPNodeManager
+		nodeManager = nm
+		cnWatcherCfgBuilder.WithNodeManager(nodeManager)
 	}
 
 	if operatorOption.Config.BGPAnnounceLBIP {
 		log.Info("Starting LB IP allocator")
 		operatorWatchers.StartLBIPAllocator(ctx, option.Config)
 	}
 
-	if operatorOption.Config.CNPNodeStatusGCInterval != 0 {
-		if err := runNodeWatcher(); err != nil {
-			log.WithError(err).Error("Unable to setup node watcher")
-		}
-	}
-
 	if kvstoreEnabled() {
```

Comment:
Looking more deeply into the `runNodeWatcher()` function, it just kicks off two async goroutines to GC CNP/CCNP node status anyway so the previous code wasn't providing any guarantees about when it would run. So should be fine from that perspective.